Update README and UG (#168)
AY2021S1-CS2103T-T12-1/tp: docs/DeveloperGuide.md

### Finding Items and Delivery

> OneShelf is capable of storing many items and pending deliveries. 
> Therefore, there is an utmost importance to have the ability to be able to find item 
> and delivery based on different fields. There could also be many similar item and 
> this will definitely benefit the user to find it quickly. <br>

Reviewer: ```suggestion
OneShelf is capable of storing many items and pending deliveries. 
Therefore, it is of utmost importance to have the ability to find items
and deliveries based on different fields. There could also be many similar items and 
this will definitely aid the user in find them quickly. <br>
```

We have modified the `find` command to be able to search for `NAME`, `SUPPLIER` 

> and `TAGS` for items using `find-i` Similarly, for deliveries, it is also possible 

Reviewer: ```suggestion
and `TAGS` for items using `find-i`. Similarly, for deliveries, it is also possible 
```

to search using the `DELIVERYNAME`, `PHONE`, `ADDRESS` or `ORDER` using `find-d`.
Note that the implementation of `find-i` and `find-d` are relatively similar and in this example, we will only show
`find-i`.

By using `ArgumentMultimap`, we are able to record the searching criteria together with the prefixes. We will then pass
 this criteria along with the prefix to create an `ItemContainsKeywordsPredicate` object which implements 
 `Predicate<Item>`.
The predicate is then passed to the `InventoryModel#UpdateItemListFilter` which will then be used to set the 
predicate on the existing filteredList.

Below is a usage example

Step 1: User execute `find-i s/NTUC` command to search the list of items by Supplier <br>

> Step 2: `ArguementMultiMap` maps each prefix to their values and `ItemFindCommandParser` checks which prefix has a 

Reviewer: user executeS. (missing s)

Author: Thanks for spotting, remedied.

### Finding Items and Delivery
OneShelf is capable of storing many items and pending deliveries. 
Therefore, it is of utmost importance to have the ability to find items
and deliveries based on different fields. There could also be many similar items and 

> this will definitely aid the user in find them quickly. <br>

Reviewer: "in finding them" or "to find them" either is fine

Reviewer: this one is my fault I apologise

### Product scope

**Target user profile**: Restaurant owners
* needs to keep track of pending deliveries
* needs to manage purchasing appropriate amounts of raw materials based on usage
* wants to be updated on raw materials stock level on a daily basis
* prefer desktop apps over other types
* can type fast
* prefers typing than mouse interactions

> * is reasonably comfortable using CLI commands

Reviewer: the original is correct (i.e. prefers typing _to_ mouse interactions)
